L/libjulia/common.jl

```diff
@@ -0,0 +1,106 @@
+    cd $WORKSPACE/srcdir/julia*
+
+    FLAGS=(
+        USE_SYSTEM_LLVM=1
+        USE_SYSTEM_LIBUNWIND=1
+        USE_SYSTEM_PCRE=1
+        USE_SYSTEM_OPENLIBM=1
+        USE_SYSTEM_DSFMT=1
+        USE_SYSTEM_BLAS=1
+        LIBBLASNAME=libopenblas
+        USE_SYSTEM_LAPACK=1
+        LIBLAPACKNAME=libopenblas
+        #USE_SYSTEM_GMP=1
+        USE_SYSTEM_MPFR=1
+        USE_SYSTEM_SUITESPARSE=1
+        USE_SYSTEM_LIBUV=1
+        USE_SYSTEM_UTF8PROC=1
+        USE_SYSTEM_MBEDTLS=1
+        USE_SYSTEM_LIBSSH2=1
+        USE_SYSTEM_CURL=1
+        USE_SYSTEM_LIBGIT2=1
+        USE_SYSTEM_PATCHELF=1
+        USE_SYSTEM_ZLIB=1
+        USE_SYSTEM_P7ZIP=1
+
+        NO_GIT=1
+        prefix="${prefix}"
+        -j${nproc}
+    )
+
+    # compile libjulia but don't try to build a sysimage
+    make "${FLAGS[@]}" julia-ui-release
+    # 'manually' install libraries and headers
+    cp usr/lib/libjulia* ${libdir}/
+    cp -R -L usr/include/julia/* ${includedir}/julia
+    install_license /usr/share/licenses/MIT
+    """
+
+    # These are the platforms we will build for by default, unless further
+    # platforms are passed in on the command line
+    platforms = expand_cxxstring_abis(supported_platforms())
+
+    # The products that we will ensure are always built
+    products = [
+        LibraryProduct("libjulia", :libjulia; dont_dlopen=true),
+    ]
+
+    # Dependencies that must be installed before this package can be built/used
+    dependencies = [
+        Dependency("LibUnwind_jll"),
+        Dependency("LibOSXUnwind_jll"),
+        Dependency(PackageSpec(name="PCRE2_jll", version=v"10.31")),
+        Dependency("OpenLibm_jll"),
+        Dependency("dSFMT_jll"),
+        Dependency(PackageSpec(name="SuiteSparse_jll", version=v"5.4.0")),
+        Dependency("LibUV_jll"),
+        Dependency("utf8proc_jll"),
+        Dependency("MbedTLS_jll"),
+        Dependency("LibSSH2_jll"),
+        Dependency("LibCURL_jll"),
+        Dependency("Patchelf_jll"),
+        Dependency("Zlib_jll"),
+        Dependency("p7zip_jll"),
+    ]
+    if version.major == 1 && version.minor == 4
+        push!(dependencies, Dependency(PackageSpec(name="OpenBLAS_jll", version=v"0.3.5")))
+        push!(dependencies, Dependency(PackageSpec(name="libLLVM_jll", version=v"8.0.1")))
+        push!(dependencies, Dependency(PackageSpec(name="MPFR_jll", version=v"4.0.2")))
+        #push!(dependencies, Dependency(PackageSpec(name="GMP_jll", version=v"6.1.2")))
+        push!(dependencies, Dependency(PackageSpec(name="LibGit2_jll", version=v"0.28.2")))
+    elseif version.major == 1 && version.minor == 5
+        push!(dependencies, Dependency(PackageSpec(name="OpenBLAS_jll", version=v"0.3.9")))
+        push!(dependencies, Dependency(PackageSpec(name="libLLVM_jll", version=v"9.0.1")))
+        push!(dependencies, Dependency(PackageSpec(name="MPFR_jll", version=v"4.1.0")))
+        #push!(dependencies, Dependency(PackageSpec(name="GMP_jll", version=v"6.1.2")))
```

Comment:
Unfortunately older GMP_jll versions can't be installed in Julia >= 1.5 and vice versa, so we can't use that unless we are willing to restrict the version(s) of Julia used to build libjulia_jll (which seems impractical?)

Reply:
We've got a solution for this now: https://github.com/JuliaPackaging/Yggdrasil/issues/1682#issuecomment-700267330